pyiron_base/jobs/job/core.py

```diff
@@ -108,6 +108,52 @@
 """
 
 
+def recursive_load_from_hdf(project_hdf5, item):
+    try:
+        group = project_hdf5[item]
+        if (
+            isinstance(group, ProjectHDFio)
+            and "NAME" in group
+            and group["NAME"] == "DataContainer"
+        ):
+            return group.to_object(lazy=True)
+        else:
+            return group
+    except ValueError:
+        pass
+
+    name_lst = item.split("/")
+
+    def successive_path_splits(name_lst):
```

Comment:
Can you move the internal function to the beginning of the function it is defined in? In this case line 112? That would help a lot with the readability.